### PR TITLE
Change withObject: to withValue:

### LINKED ISSUE #1585: Rename initWithObject/createWithObject to withValue
We need to make this change in objective-c to be consistent with the new swift apis and to make things more clear/less confusing. 

This change should be made once al-swift is merged to master, at which point in time we can fix Object.init(object:)

Old api calls should be deprecated for a few releases.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1585.

\c @mrackwitz @kishikawakatsumi

### CODE REVIEW

**Comment by jpsim** [comment]
The docs have a number of outstanding `@param object`s that will need to be converted to `@param value`.

**Comment by kishikawakatsumi** [comment]
LGTM

**Comment by segiddins** [comment]
@jpsim would appreciate a quick look at the CHANGELOG?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,16 @@ x.xx.x Release notes (yyyy-MM-dd)
 ### API breaking changes
 
 * Migration blocks are no longer called when a Realm file is first created.
+* The following APIs have been deprecated in favor of newer method names:
+
+| Deprecated API | New API |
```
Comment (jpsim) [inline CHANGELOG.md]: Could you align this to be legible in plaintext?

File: Realm/RLMMigration.h [hunk 2]
```diff
@@ -69,14 +69,16 @@ typedef void (^RLMObjectMigrationBlock)(RLMObject *oldObject, RLMObject *newObje
...
                     object, or a JSON object such as those returned from the methods in NSJSONSerialization, or
                     an NSArray with one object for each persisted property. An exception will be
                     thrown if any required properties are not present and no default is set.
 
                     When passing in an NSArray, all properties must be present, valid and in the same order as the properties defined in the model.
  */
--(RLMObject *)createObject:(NSString *)className withObject:(id)object;
+-(RLMObject *)createObject:(NSString *)className withValue:(id)value;
+
+-(RLMObject *)createObject:(NSString *)className withObject:(id)object DEPRECATED_MSG_ATTRIBUTE("use createObject:WithValue:");
```
Comment (jpsim) [inline Realm/RLMMigration.h]: `use createObject:withValue:`

**Comment by jpsim** [comment]
:+1: other than my two very minor changes.

**Comment by segiddins** [comment]
@jpsim done!

**Pull request merged**